aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -326,6 +360,123 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    public entry fun rotate_authentication_key_with_rotation_capability(
+        delegate_signer: &signer,
+        rotation_cap_offerer_address: address,
+        new_scheme: u8,
+        new_public_key_bytes: vector<u8>,
+        cap_update_table: vector<u8>
+    ) acquires Account, OriginatingAddress {
+        assert!(exists_at(rotation_cap_offerer_address), error::not_found(EOFFERER_ADDRESS_DOES_NOT_EXIST));
+
+        // check that there exists a rotation capability offer at the offerer's account resource for the delegate
+        let offerer_account_resource = borrow_global<Account>(rotation_cap_offerer_address);
+        let delegate_address = signer::address_of(delegate_signer);
+        assert!(option::contains(&offerer_account_resource.rotation_capability_offer.for, &delegate_address), error::not_found(ENO_SUCH_ROTATION_CAPABILITY_OFFER));
+
+        let delegate_account_resource = borrow_global<Account>(delegate_address);
+        let curr_auth_key = from_bcs::to_address(offerer_account_resource.authentication_key);
+        let challenge = RotationProofChallenge {
+            sequence_number: delegate_account_resource.sequence_number,
+            originator: rotation_cap_offerer_address,
+            current_auth_key: curr_auth_key,
+            new_public_key: new_public_key_bytes,
+        };
+
+        // verify that `cap_update_table` has the correct `RotationProofChallenge` and is signed by the new private key
```

Comment:
The phrasing here is rather unprecise: `cap_update_table` is a signature, so it's unclear what it means for "a signature to have a correct RotationProofChallenge` (even though one might be able to guess).

A clearer comment would be:
```
// Verifies that the `RotationProofChallenge` from above is signed under the new public key that we are rotating to.
```